plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -114,10 +118,12 @@ public class ThriftHiveMetastoreClient
     private final AtomicInteger chosenGetTableAlternative;
     private final AtomicInteger chosenAlterTransactionalTableAlternative;
     private final AtomicInteger chosenAlterPartitionsAlternative;
+    private final String catalogName;
```

Comment:
```suggestion
private final Optional<String> catalogName;
```

Advocating here for prepending the catalog to the database name only if it is necessary.

If the catalog name is `hive` (default) we don't need to change any existing payloads sent to Hive.

Reply:
oh ok, should we make it Optional and checks for the "hive"?
I felt keeping "hive" as the default catalog simplifies the code, and we dont need to check it while making the request.